src/python_testing/TC_ACL_2_11.py

```diff
@@ -118,16 +118,16 @@ async def test_TC_ACL_2_11(self):
                 restriction_type = restriction.type
                 ID1 = restriction.id
 
-                attribute = ALL_ATTRIBUTES[C1][ID1]
-                command = ALL_ACCEPTED_COMMANDS[C1][ID1]
-
                 if restriction_type == AccessControl.Enums.AccessRestrictionTypeEnum.kAttributeAccessForbidden:
+                    attribute = ALL_ATTRIBUTES[C1][ID1]
                     await self.read_single_attribute_expect_error(cluster=cluster, attribute=attribute, error=Status.AccessRestricted, endpoint=E1)
                 elif restriction_type == AccessControl.Enums.AccessRestrictionTypeEnum.kAttributeWriteForbidden:
+                    attribute = ALL_ATTRIBUTES[C1][ID1]
                     status = await self.write_single_attribute(attribute_value=attribute, endpoint_id=E1)
                     asserts.assert_equal(status, Status.AccessRestricted,
                                          f"Failed to verify ACCESS_RESTRICTED when writing to Attribute {ID1} Cluster {C1} Endpoint {E1}")
                 elif restriction_type == AccessControl.Enums.AccessRestrictionTypeEnum.kCommandForbidden:
+                    command = ALL_ACCEPTED_COMMANDS[C1][ID1]
                     result = await self.send_single_cmd(cmd=command, endpoint=E1)
```

Comment:
Does this send the command if the command format is completely wrong? Do you have a log showing that the command got sent?

Also, if the element is a wildcard, you have to iterate over all known commands of the cluster, not just `ID1`, which would otherwise fail.

Reply:
yes, fortunately the write attribute and invoke command operations do go over the wire and get evaluated for access before the arguments/values are validated.

Reply:
wildcard support fixed.